### DISCUSSION TITLE
Nested Subgraphs

### DISCUSSION THREAD

**Discussion by AsdfplayaV2** [mermaid-js/mermaid#4291] [Q&A] [answered]
Hello,

i'm trying to create flowcharts with multiple nested subgraphs but for some reason something in the syntax seems wrong. And i tested some other variations and got it to work, but i can't really grasp why it works.
Some Examples: 
So what i want to do is this

flowchart TB
subgraph A
subgraph B
subgraph C
subgraph D
p1-->p2
p2-->p3
p3-->E
subgraph E
p4[1]
end
end
end
end
end

but the resulting flowchart is none and i get a syntax error in graph 
https://mermaid.live/edit#pako:eNptzz0LwkAMBuC_cmRuh1anDoJaRyed9BxiL7XF3gfXO0RK_7tREW5wCQ_hheSdoLGKoIJ2sI-mQx_EcSPNGK83j64T68Tpfpu4lsYVeb5yJaN8Y8FYMHZJiu2W5-IiDRn1b0AGmrzGXvE7kzRCSAgdaZJQMRX6uwRpZs5hDPbwNA1UwUfKIDqFgeoe-ZT-LUn1wfr9t96nZQYOzclajrQ4jDS_AHgAWKU

So i kinda got it to work after adding a Process Node in subgraph A or subgraph B

flowchart TB
subgraph A
p0
subgraph B
subgraph C
subgraph D
p1-->p2
p2-->p3
p3-->E
subgraph E
p4[1]
end
end
end
end
end

https://mermaid.live/edit#pako:eNptjz0LwjAQhv9KuLkF2zplENQ6OumkcTib1BabD9IEkdL_7qkIGVyO5-4euHsnaKxUwKEd7KPp0Ad23AgzxuvNo-vYWhi3SPp0t024Jq_I85UrCco3VAQVwS6xiN3yXFyEUUb-K5CBVl5jL-mlSRjGBIROaSWAE0r0dwHCzORhDPbwNA3w4KPKIDqJQdU90ikNvMVhpKmSfbB-_834iZqBQ3Oy9ufMLzt5Wlo

It also works if i declare the subgraph id in the previous subgraph

flowchart TB
subgraph A
subgraph B
subgraph C
subgraph D
E
p1-->p2
p2-->p3
p3-->E
subgraph E
p4[1]
end
end
end
end
end

https://mermaid.live/edit#pako:eNptjzEPgjAQhf8KuRkGwInBRMXRSSc9h5MWIdJrU9oYQ_jvFo1JB5fLl5cvefcmaLSQUEE76GfTkXXJaYs8-tvdkumSTcRxvou4Rt4jmzzL1qYIUCxQBigD7CNxsVaX_IosWfw7kIKSVlEvwkcTcpIguE4qiVAFFGQfCMhz8Mg7fXxxA5WzXqbgjSAn655ClYKqpWEMqRS90_bwnfhZmoIhPmv9c-Y34eJZ_w

But if possible i'd really like to use my first code. So is this a problem with my Syntax or from Mermaid?

**Comment by elzup** [comment, marked as answer]
It's strange

It seems dagre(renderer)'s probrem https://github.com/dagrejs/dagre/issues/236
error is `Cannot set properties of undefined (setting 'rank')`


with elk works (use `%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%`)

```mermaid
%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%

flowchart TB
subgraph A
subgraph B
subgraph C
subgraph D
p1-->p2
p2-->p3
p3-->E
subgraph E
p4[1]
end
end
end
end
end
```

**Reply by AsdfplayaV2** [reply]
Thank you, this gets the original syntax to work.
P.S. do you know how to style specific IDs similar to dagre in elk?
e.g.

flowchart TB 
subgraph A
end
style A fill:#FDF
